rts/Sim/Misc/ExitOnlyMap.h

```diff
@@ -0,0 +1,50 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+#ifndef EXIT_ONLY_MAP_H
+#define EXIT_ONLY_MAP_H
+
+// #include <array>
+#include <cstdint>
+#include <vector>
+
+#include "System/Log/ILog.h"
+
+class ExitOnlyMap {
+public:
+    static constexpr int resolution = 2;
+    static constexpr int sectorWidth = 16;
+    static constexpr int internalSectorWidth = sectorWidth / resolution;
+
+    uint32_t interleave(uint32_t x0, uint32_t y0)
+    {
+        static const uint32_t B[] = {0x0000FFFF, 0x00FF00FF, 0x0F0F0F0F, 0x33333333, 0x55555555};
+        static const unsigned S[] = {16, 8, 4, 2, 1};
+
+        uint32_t x = x0;
+        uint32_t y = y0;
+
+        for(unsigned i = 0; i < sizeof(B)/sizeof(B[0]); i++)
+        {
+            x = (x | (x << S[i])) & B[i];
+            y = (y | (y << S[i])) & B[i];
```

Comment:
must `S[i]` always be `1 << i`? maybe it doesn't need to be a separate array if so

Reply:
It isn't `1 << i` though. The shifts are going down in scale, not up.

Reply:
`1 << (B.size() - i - 1)` then?